addons/stock/models/product.py

```diff
@@ -799,25 +802,28 @@ def _compute_product_tooltip(self):
     def _onchange_tracking(self):
         return self.mapped('product_variant_ids')._onchange_tracking()
 
+    @api.depends('type')
+    def _compute_tracking(self):
+        self.filtered(
+            lambda t: not t.tracking or t.type == 'consu' and t.tracking != 'none'
```

Comment:
The `and t.tracking != 'none' is probably not necessary, unless writing the same value triggers a lot of other computes :thinking: